tests/bwrite.rs

```diff
@@ -17,128 +17,554 @@
 pub mod util;
 
 use assert_cmd::prelude::*; // Add methods on commands
+use base64::{engine::general_purpose, Engine as _};
 use predicates::prelude::*; // Used for writing assertions
 use serde_json::Value;
-use std::collections::HashSet;
 use std::fs::File;
 use std::io::Write;
 use std::path::PathBuf;
 use tempfile::Builder;
 
 #[tokio::test]
-async fn test_batch_write() -> Result<(), Box<dyn std::error::Error>> {
+async fn test_batch_write_json_put() -> Result<(), Box<dyn std::error::Error>> {
     let mut tm = util::setup().await?;
     let table_name = tm.create_temporary_table("pk", None).await?;
 
-    let mut path = PathBuf::from(env!("CARGO_MANIFEST_DIR"));
-    path.push("tests/resources/test_batch_write.json");
-    let test_json_content = std::fs::read_to_string(path)?;
+    let tmpdir = Builder::new().tempdir()?; // defining stand alone variable here as tempfile::tempdir creates directory and deletes it when the destructor is run.
+    let batch_input_file_path = create_test_json_file(
+        "tests/resources/test_batch_write_put.json",
+        vec![&table_name],
+        &tmpdir,
+    );
+
+    let mut c = tm.command()?;
+    c.args(&[
+        "--region",
+        "local",
+        "bwrite",
+        "--input",
+        &batch_input_file_path,
+    ])
+    .output()?;
+
+    let mut c = tm.command()?;
+    let scan_cmd = c.args(&[
+        "--region",
+        "local",
+        "--table",
+        &table_name,
+        "scan",
+        "-o",
+        "raw",
+    ]);
+    let output = scan_cmd.output()?.stdout;
+    // Catch an error if key does not exist
+    let mut output_json: Value =
+        serde_json::from_str::<serde_json::Value>(&String::from_utf8(output)?)?
+            .get(0)
+            .unwrap()
+            .clone();
+
+    let org_json_string = std::fs::read_to_string("tests/resources/test_batch_write_put.json")?;
+    // Catch an error if key does not exist
+    let mut org_json = serde_json::from_str::<serde_json::Value>(&org_json_string)?
+        .get("__TABLE_NAME__1")
+        .unwrap()
+        .get(0)
+        .unwrap()
+        .get("PutRequest")
+        .unwrap()
+        .get("Item")
+        .unwrap()
+        .clone();
```

Comment:
Can we prepare a JSON file to validate the result? Capturing meaning is hard at a glance.

Reply:
Created a new expected json file.